rfcs/experimental/kernel_templates/README.md

```diff
@@ -309,10 +309,37 @@ as described in [Abstract Algorithm Signature](abstract-algorithm-signature) sec
 In particular, it should be investigated if supporting these entities is necessary and
 whether they can be handled by an additional thin layer between
 the algorithms with standard interfaces and Kernel Templates:
-- `oneapi::dpl::begin` and `oneapi::dpl::end`, which mimic iterators to pass `sycl::buffer` to algorithms accepting iterators.
+- `oneapi::dpl::begin` and `oneapi::dpl::end`,
+   which mimic iterators to pass `sycl::buffer` to iterator-based algorithms.
 - Iterators to a host-allocated `std::vector` as described in
   [Pass Data to Algorithms > Use std::vector](https://github.com/uxlfoundation/oneDPL/blob/release/2022.9.0/documentation/library_guide/parallel_api/pass_data_algorithms.rst#use-stdvector)
 
+An additional question is how to handle hardware requirements that cannot be queried directly
+through standard SYCL mechanisms. For example, selecting a more performant kernel template
+might depend on forward-progress guarantees between work-groups.
+While oneAPI DPC++ supports querying this via
+the [Root Group](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_root_group.asciidoc)
+experimental extension, other SYCL implementations may lack a corresponding feature.
```

Comment:
That's not a problem I think - at least not for the use of KT as a backend. Implementations with HW requirements like forward progress are supposed to have a fallback path - either another KT algorithm or a "standard SYCL" implementation. If we cannot prove that the requirements are met for any reason, whether HW or SYCL implementation related, the fallback should be used.